src/Operation/DeleteProduct.php

```diff
@@ -90,6 +90,9 @@ public function delete()
             $this->account->getName(),
             $this->activeDomain
         );
+        if (empty($this->productIds)) {
```

Comment:
Why not check before the `initRequest`?

Reply:
I was thinking about that as well but I though that if the `initRequest` fails the error should be thrown. Otherwise the request init issues might go unnoticed.

Reply:
Good point!